sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinHelper.scala

```diff
@@ -303,6 +303,56 @@ object StreamingSymmetricHashJoinHelper extends Logging {
     }
   }
 
+  /**
+   * A custom RDD that allows partitions to be "zipped" together, while ensuring the tasks'
+   * preferred location is based on which executors have the required join state stores already
+   * loaded. This class is a variant of [[org.apache.spark.rdd.ZippedPartitionsRDD2]] which only
+   * changes signature of `f` by taking in a map of column family schemas. This is used for
+   * passing the column family schemas when there is initial state for the TransformWithStateExec
+   * operator
+   */
+  class StateStoreAwareZipPartitionsRDDWithSchemas[A: ClassTag, B: ClassTag, V: ClassTag](
```

Comment:
Is it better to pass this as a planning rule or should we change the mapPartitions interface ?

Reply:
lets remove these ?